lib/streamlit/external/langchain/streamlit_callback_handler.py

```diff
@@ -256,7 +258,7 @@ def complete(self, final_label: str | None = None) -> None:
 
         self._container.update(
             label=final_label,
-            expanded=False if self._collapse_on_complete else None,
+            expanded=False if self._collapse_on_complete else True,
```

Comment:
I believe the reason why we used `None` here is to keep the expander at whatever state it is on the frontend. Since the user can open and close the expander on the frontend, and we don't know in the (python) backend whats state its at, we are just sending out `None`. This tells the frontend to keep it at whatever state it currently is. 

I haven't tested it, but I think it should stay open when it's open on the front end. Or isn't that the case in your test?

Reply:
In my local testing, it will always collapse the message window.  I believe somewhere there is a `if expanded`, and `None` is treated as False in Python.

Reply:
I'm not able to reproduce this issue if its set to `None` with this code:

```
import streamlit as st
from streamlit.external.langchain import StreamlitCallbackHandler
from langchain.llms.openai import OpenAI
from langchain.chains import LLMChain
from langchain.prompts import PromptTemplate
from langchain_openai import OpenAI


def load_chain():
    """Logic for loading the chain you want to use should go here."""
    llm = OpenAI(
        openai_api_key="<KEY>",
        streaming=True,
    )
    template = """Question: {question}

    Answer: Let's think step by step."""

    prompt = PromptTemplate.from_template(template)

    llm_chain = LLMChain(prompt=prompt, llm=llm)
    return llm_chain


if prompt := st.chat_input():
    llm_chain = load_chain()
    with st.chat_message("🧞‍♂️"):
        cbh = StreamlitCallbackHandler(st.container())
        llm_chain.run(prompt, callbacks=[cbh])
```

Uploading Screen Recording 2024-03-24 at 22.03.42.mov…

Reply:
What type of chain do you use to test this?

Reply:
I am using the following one for my RAG application:

```
RetrievalQA.from_chain_type(
                llm=self.llm,
                chain_type="stuff",
                retriever=self.vector_db.as_retriever(),
                memory=self.memory,
                return_source_documents=self.return_source_documents,
            )
```

This is a side issue and I can monkey-patch it on the application side. I will revert the changes to help get the fix in faster.